### PR TITLE
[stag] PLT-302 Optimize Persona ES alias

### PR DESCRIPTION
**Pull request by nikhilbonte21**

### CODE REVIEW

File: repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java [hunk 1]
```diff
@@ -178,19 +184,27 @@ private void personaPolicyToESDslClauses(List<AtlasEntity> policies,
...
-                            addPersonaGlossaryFilterClauses(glossaryQName, allowClauseList);
+                            terms.add(glossaryQName);
+                            allowClauseList.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, "*@" + glossaryQName)));
                         }
                     }
                 }
             }
         }
+
+        if (terms.size() > assetsMaxLimit) {
```
Comment (checkaayush) [inline repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java:203]: Just an optimization thought: I'm not sure about what `terms.size()` can balloon up to, but should we consider throwing exception sooner than end of preparation for `terms`, maybe as an `if` clause somewhere in the loop throw exception as soon as we reach `assetsMaxLimit`.

**Review by checkaayush — APPROVED** [review]
Left a minor comment. Rest LGTM.

File: repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java (hunk 1 above)
Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java]: Addressed

Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java]: terms.size() + assets.size() + 1 for Metadata policies where along with assets we add 1 more clause for connection qualifiedName hence +1

For Glossary policies, it is only the assets count.

File: repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java [hunk 2]
```diff
@@ -165,32 +170,46 @@ private Map<String, Object> getFilterForPurpose(AtlasEntity purpose) throws Atla
...
+        List<String> terms = new ArrayList<>();
         for (AtlasEntity policy: policies) {
 
             if (policy.getStatus() == null || AtlasEntity.Status.ACTIVE.equals(policy.getStatus())) {
                 List<String> assets = getPolicyAssets(policy);
 
                 if (getIsAllowPolicy(policy)) {
                     if (getPolicyActions(policy).contains(ACCESS_READ_PERSONA_METADATA)) {
+
+                        if (terms.size() + assets.size() + 1 > assetsMaxLimit) {
```
Comment (checkaayush) [inline repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java:182]: nit: There should be a comment here clarifying the "why" behind this clause (why +1, etc.).

**Review by checkaayush — APPROVED** [review]

File: repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java (hunk 2 above)
Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java:182]: Makes sense, I will do that

**Review by checkaayush — APPROVED** [review]

**Pull request merged**